.github/actions/setup-test-environment/action.yml

```diff
@@ -0,0 +1,36 @@
+name: 'Set up test environment'
+description: 'Set up test environment for spectral-workbench'
+
+services:
+  mysql:
+    image: mysql:5.7
+    env:
+      MYSQL_ROOT_PASSWORD: root
+    ports:
+    - 3306
+    options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+runs:
+  using: "composite"
+  steps:
+
+  - name: Install JavaScript dependencies with Yarn
+    shell: bash
+    run: yarn check || yarn install --frozen-lockfile;
+
+  - name: Setup database
+    shell: bash
+    env:
+      RAILS_ENV: test
+      DB_PASSWORD: root
+      # tell Rails to use proper port for MySQL
+      DB_PORT: ${{ job.services.mysql.ports[3306] }}
+      DISABLE_BOOTSNAP: true # bootsnap cache failing when multiple parallel builds try to access it at the same time
+    run: |
+      cp config/database.yml.example config/database.yml
+      cp db/schema.rb.example db/schema.rb
+      cp config/config.yml.example config/config.yml
+      sudo systemctl start mysql
+      mysql -uroot -proot -e "CREATE DATABASE swbench_test;"
+      mysql -uroot -proot -e "GRANT ALL PRIVILEGES ON *.* TO 'swbench_test'@'%';";
```

Comment:
I tried this because i got:

`Mysql2::Error::ConnectionError: Access denied for user 'runner'@'localhost' (using password: NO)`

but after this i got:

`ERROR 1410 (42000) at line 1: You are not allowed to create a user with GRANT`